### PR TITLE
parser: fix LOCK clause error message

### PR DESCRIPTION
**Pull request by lauhg**
#### Fix compatibility problem about `LOCK` syntax

Issue: pingcap/parser#508

MySQL Syntax:
```sql
alter_lock_option:
          LOCK_SYM opt_equal alter_lock_option_value { $$= $3; }
        ;

alter_lock_option_value:
          DEFAULT_SYM
          {
            $$= Alter_info::ALTER_TABLE_LOCK_DEFAULT;
          }
        | ident
          {
            if (is_identifier($1, "NONE"))
              $$= Alter_info::ALTER_TABLE_LOCK_NONE;
            else if (is_identifier($1, "SHARED"))
              $$= Alter_info::ALTER_TABLE_LOCK_SHARED;
            else if (is_identifier($1, "EXCLUSIVE"))
              $$= Alter_info::ALTER_TABLE_LOCK_EXCLUSIVE;
            else
            {
              my_error(ER_UNKNOWN_ALTER_LOCK, MYF(0), $1.str);
              MYSQL_YYABORT;
            }
          }
        ;
```

Bad SQL Case:
```sql
alter table t lock = first
alter table t lock = start
alter table t lock = commit
alter table t lock = binlog
```

### Check List 

Tests 

 - Unit test

### CODE REVIEW

File: parser.y [hunk 1]
```diff
@@ -1738,26 +1738,24 @@ AlgorithmClause:
...
-		$$ = ast.LockTypeExclusive
-	}
-|	"LOCK" EqOpt identifier
+|	"LOCK" EqOpt Identifier
 	{
-		yylex.AppendError(ErrUnknownAlterLock.GenWithStackByArgs($3))
-		return 1
+		id := strings.ToUpper($3)
+
+		if id == "NONE" {
```
Comment (leoppro) [inline parser.y]: how about handle `Default` here?

Comment (lauhg) [reply parser.y]:
```go
	"LOCK" EqOpt "DEFAULT"
	{
		$$ = ast.LockTypeDefault
	}
```
See https://github.com/pingcap/parser/pull/516/files#diff-966f3247252fb3ffd7810a3b943884f0R1741

Comment (leoppro) [reply parser.y]: I mean that we remove `"LOCK" EqOpt "DEFAULT"` rule and add a branch `if id == "DEFAULT"` in line 1749

Comment (lauhg) [reply parser.y]: Oh, I misread it.
`"DEFAULT"` is reserved keyword, it's not included in `Identifier`.
```go
Identifier:
identifier | UnReservedKeyword | NotKeywordToken | TiDBKeyword
```

Comment (leoppro) [reply parser.y]: fine LGTM

**Comment by leoppro** [comment]
LGTM

**Pull request merged**